src/GovUk.Education.ExploreEducationStatistics.Notifier/GovUk.Education.ExploreEducationStatistics.Notifier.csproj

```diff
@@ -28,9 +28,5 @@
     <None Update="host.json">
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </None>
-    <None Update="local.settings.json">
-      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
-      <CopyToPublishDirectory>Never</CopyToPublishDirectory>
-    </None>
```

Comment:
Guess you might not have noticed, but in our other function apps, we have startups that load in `appsettings.Local.json` files that are .gitignored and won't be checked into source control.

We can enable loading of this config file by simply adding:

```cs
public class Startup : FunctionsStartup
{
    public override void ConfigureAppConfiguration(IFunctionsConfigurationBuilder builder)
    {
        var binDir = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
        var rootDir = Path.GetFullPath(Path.Combine(binDir!, ".."));
        builder.ConfigurationBuilder
            .AddJsonFile($"{rootDir}/appsettings.Local.json", optional: true, reloadOnChange: true);
    }
}
```

I think this should get merged together with `local.settings.json` so it should get the behaviour you're looking for.

Would suggest trying this out before going with the proposed approach. It's not ideal to add additional set up steps as this complicates things for anyone coming onto the project.

For most of the project, we've adopted sensible defaults so that things usually work out of the box after the main set up steps in the project README have been completed.

Reply:
@ntsim In #4673 I've adopted the Options pattern for our custom settings.

### local.settings.json

I've kept `local.settings.json` as this seems to be a standard for running functions locally for all example functions projects I've seen, and this has the core properties required to configure a function app.

`local.settings.json` now only has additional properties for `BaseUrl`, `TableStorageConnectionString`, `TokenSecretKey`, `PublicAppUrl`. These are all defaults which should never need to be changed when running locally.

```
{
  "IsEncrypted": false,
  "Values": {
    "AzureWebJobsStorage": "UseDevelopmentStorage=true;DevelopmentStorageProxyUri=http://data-storage",
    "FUNCTIONS_WORKER_RUNTIME": "dotnet-isolated",
    "AppSettings:BaseUrl": "http://localhost:7073/api",
    "AppSettings:TableStorageConnectionString": "AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;DefaultEndpointsProtocol=http;QueueEndpoint=http://data-storage:10001/devstoreaccount1;TableEndpoint=http://data-storage:10002/devstoreaccount1;",
    "AppSettings:TokenSecretKey": "email-token-security-signing-key",
    "AppSettings:PublicAppUrl": "http://localhost:3000"
  },
  "ConnectionStrings": {
  },
  "Host": {
    "LocalHttpPort": 7073,
    "CORS": "*"
  }
}
```

### appsettings.Local.json

I've added additional config during Notifier startup to load the gitignored `appsettings.Local.json` like you mentioned and this is now consistent with our other function apps.

`appsettings.Local.json` can now be used for all the gitignored GovUkNotify config and I've added a `appsettings.Local.json.example` file as follows:

```
{
  "GovUkNotify" : {
    "ApiKey": "",
    "EmailTemplates" : {
      "ReleaseAmendmentPublishedId": "",
      "ReleaseAmendmentPublishedSupersededSubscribersId": "",
      "ReleasePublishedId": "",
      "ReleasePublishedSupersededSubscribersId": "",
      "SubscriptionConfirmationId": "",
      "SubscriptionVerificationId": ""
    }
  }
}
```

If you agree with this then the changes on this PR except for the README could be scrapped when #4673 is merged? I can alter the README here to explain setting up `appsettings.Local.json`.

Reply:
Chatted with @ntsim yesterday and there's a slight change to what I proposed above. Each function app now has an `appsettings.json` for our own app variables which don't need to change and has an `appsettings.Local.json` which is gitignored for overrides and secrets.